### PR TITLE
Add python-ldap module to conditional requirements

### PR DESCRIPTION
**Pull request by abretaud**
When using the ldap auth method, the python-ldap module is required

### CODE REVIEW

File: lib/galaxy/dependencies/__init__.py [hunk 1]
```diff
@@ -76,6 +89,9 @@ def check_pygments( self ):
         # pygments is a dependency of weberror and only weberror
         return self.check_weberror()
 
+    def check_python_ldap( self ):
+        return 'ldap' in self.authenticators
```
Comment (nsoranzo) [inline lib/galaxy/dependencies/__init__.py]: `or 'activedirectory' in self.authenticators`

**Comment by nsoranzo** [comment]
Thanks @abretaud, that's sweet!

**Comment by abretaud** [comment]
Somewhat related thing: do you know if we can have access to the auth method(s) from templates? The "Forgot password?" link at https://github.com/galaxyproject/galaxy/blob/dev/templates/user/login.mako#L91 sometimes confuses our users as it as no effect with ldap auth. It would be nice to hide it in this case

**Comment by nsoranzo** [comment]
@abretaud It may be technically possible, but I don't think that's a good idea because you may have multiple authentication available at the same time (e.g. LDAP for internal users and Galaxy password database for external ones).

**Comment by abretaud** [comment]
Oh yes, I see the point. Maybe it could be hidden only for simple cases (e.g. when there is only ldap auth configured)? Anyway, it's just a cosmetic detail, so no worry

**Comment by hexylena** [comment]
@nsoranzo is this ready to merge?

**Comment by bgruening** [comment]
I have hope this will fix this one too: https://github.com/bgruening/docker-galaxy-stable/pull/199

**Comment by nsoranzo** [comment]
@erasche I think we have to wait for https://github.com/galaxyproject/starforge/pull/97 to be merged and for the wheels to be online.

**Comment by hexylena** [comment]
@nsoranzo ok, thanks, just following up on a low hanging fruit. :)

**Comment by dannon** [comment]
Merged this to get it in, we can follow up on the wheel on the starforge side.  This just enables the conditional dependency, which will force this to be built/installed from pypi until we get the wheel up, which is not a bad thing.

File: lib/galaxy/dependencies/conditional-requirements.txt [hunk 2]
```diff
@@ -10,3 +10,4 @@ pbs_python
 drmaa
 statsd
 # PyRods not in PyPI
+python-ldap==2.4.25
```
Comment (nsoranzo) [inline lib/galaxy/dependencies/conditional-requirements.txt:13]: A 2.4.26 release came out in the mean time, if you want to update here.

**Pull request merged**